clients/ui/frontend/src/app/pages/modelRegistry/screens/RegisterModel/RegisterModel.tsx

```diff
@@ -61,31 +85,50 @@ const RegisterModel: React.FC = () => {
         <Form isWidthLimited>
           <Stack hasGutter>
             <StackItem className={spacing.mbLg}>
-              <PrefilledModelRegistryField mrName={mrName} />
+              <FormGroup
+                className="form-group-disabled"
+                label="Model registry"
+                isRequired
+                fieldId="mr-name"
+              >
+                <div className="form-fieldset-wrapper">
+                  {modelRegistryInput}
+                  <fieldset aria-hidden="true" className="form-fieldset">
+                    <legend className="form-fieldset-legend">
+                      <span>Model Registry</span>
+                    </legend>
+                  </fieldset>
+                </div>
+              </FormGroup>
             </StackItem>
             <StackItem>
               <FormSection
                 title="Model details"
                 description="Provide general details that apply to all versions of this model."
               >
                 <FormGroup label="Model name" isRequired fieldId="model-name">
-                  <TextInput
-                    isRequired
-                    type="text"
-                    id="model-name"
-                    name="model-name"
-                    value={modelName}
-                    onChange={(_e, value) => setData('modelName', value)}
-                  />
+                  <div className="form-fieldset-wrapper">
+                    {modelNameInput}
+                    <fieldset aria-hidden="true" className="form-fieldset">
+                      <legend className="form-fieldset-legend">
+                        <span>Model Name</span>
+                      </legend>
+                    </fieldset>
+                  </div>
                 </FormGroup>
-                <FormGroup label="Model description" fieldId="model-description">
-                  <TextArea
-                    type="text"
-                    id="model-description"
-                    name="model-description"
-                    value={modelDescription}
-                    onChange={(_e, value) => setData('modelDescription', value)}
-                  />
+                <FormGroup
+                  className="model-description"
+                  label="Model description"
+                  fieldId="model-description"
+                >
+                  <div className="form-fieldset-wrapper">
+                    {modelDescriptionInput}
+                    <fieldset aria-hidden="true" className="form-fieldset">
+                      <legend className="form-fieldset-legend">
+                        <span>Model Description</span>
+                      </legend>
+                    </fieldset>
+                  </div>
```

Comment:
Why does this structure need to be added? Is this how we get the label to get inserted into the TextInput? Not against anything here just looking to learn.

Reply:
Great question! Yes, this applies the nested label within the input border, which couldn't be achieved with pure CSS using PF's default component structure. MUI uses `fieldset` to achieve this styling which we applied here to get the same look and feel on the text fields.